libs/order-list/src/lib/components/mocks/generate-orders.ts

```diff
@@ -0,0 +1,156 @@
+import merge from 'lodash/merge';
+import type { PartialDeep } from 'type-fest';
+
+import {
+  OrderStatus,
+  OrderTimeInForce,
+  OrderType,
+  Side,
+} from '@vegaprotocol/types';
+import type { Orders, Orders_party_orders } from '../__generated__/Orders';
+
+export const generateOrders = (override?: PartialDeep<Orders>): Orders => {
+  const orders: Orders_party_orders[] = generateOrdersArray();
+
+  const defaultResult = {
+    party: {
+      id: 'party-id',
+      orders,
+      __typename: 'Party',
+    },
+  };
+  return merge(defaultResult, override);
+};
+
+export const generateOrder = (partialOrder: Partial<Orders_party_orders>) =>
+  merge(
+    {
+      __typename: 'Order',
+      id: 'order-id2',
+      market: {
+        __typename: 'Market',
+        id: 'market-id',
+        name: 'market-name',
+        decimalPlaces: 2,
+        positionDecimalPlaces: 2,
+        tradableInstrument: {
+          __typename: 'TradableInstrument',
+          instrument: {
+            __typename: 'Instrument',
+            code: 'instrument-code',
+          },
+        },
+      },
+      size: '10',
+      type: OrderType.Market,
+      status: OrderStatus.Active,
+      side: Side.Buy,
+      remaining: '5',
+      price: '',
+      timeInForce: OrderTimeInForce.IOC,
+      createdAt: new Date().toISOString(),
+      updatedAt: null,
+      expiresAt: null,
+      rejectionReason: null,
+    } as Orders_party_orders,
+    partialOrder
+  );
+
+export const limitOrder = generateOrder({
+  id: 'limit-order',
+  type: OrderType.Limit,
+  status: OrderStatus.Active,
+  timeInForce: OrderTimeInForce.GTT,
+  createdAt: new Date('2022-3-3').toISOString(),
+  expiresAt: new Date('2022-3-5').toISOString(),
+});
+
+export const marketOrder = generateOrder({
+  id: 'market-order',
+  type: OrderType.Market,
+  status: OrderStatus.Active,
+});
+
+export const generateMockOrders = (): Orders_party_orders[] => {
+  return [
+    generateOrder({
```

Comment:
nitpick (non-blocking): lot of duplicated fields here, might be nice to have a default object and spread that into these